pkg/apitype/migrate/resource.go

```diff
@@ -47,3 +48,31 @@ func UpToResourceV2(v1 apitype.ResourceV1) apitype.ResourceV2 {
 	v2.InitErrors = append(v2.InitErrors, v1.InitErrors...)
 	return v2
 }
+
+// UpToResourceV3 migrates a resource from ResourceV2 to ResourceV3.
+func UpToResourceV3(v2 apitype.ResourceV2) apitype.ResourceV3 {
+	var v3 apitype.ResourceV3
+	v3.URN = v2.URN
+	v3.Custom = v2.Custom
+	v3.Delete = v2.Delete
+	v3.ID = v2.ID
+	v3.Type = v2.Type
+	v3.Inputs = v2.Inputs
```

Comment:
In general we're super defensive in this package and do deep copies anywhere there's any chance for state to be shared between the passed-in apitype and the returned apitype. I tend to be paranoid so I have done a lot of deep copies in this package; I'm fine not doing so as long as we are absolutely sure that we're not accidentally shooting ourselves in the foot by sharing input/output maps and dependency slices between apitypes.